### PR TITLE
Update card pattern to support color themes

### PR DESCRIPTION
**Pull request by jmuzina**
## Done

Updates the card component to use new color theme variables & thus support theming.

Note that the image used in the header example was changed as the [old one](https://vanillaframework.io/docs/examples/patterns/card/header) is not friendly to all themes (dark text becomes impossible to read on dark theme)

Fixes [WD-11865](https://warthogs.atlassian.net/browse/WD-11865)

## QA

- Review [card docs](https://vanilla-framework-5112.demos.haus/docs/patterns/card) & verify cards appear as expected.
- Review all card examples and verify they look as expected in all color themes.
    - [Default](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/default?theme=light)
    - [Content bleed](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/content-bleed?theme=light)
    - [Header](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/header?theme=light)
    - [Highlighted](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/highlighted?theme=light)
    - [Image](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/image?theme=light)
    - [Overlay](https://vanilla-framework-5112.demos.haus/docs/examples/patterns/card/overlay?theme=light)
- Review [4.13.0 release notes](https://vanilla-framework-5112.demos.haus/docs/whats-new)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

Default light:
image
Default dark:
image
Default paper:
image

Content bleed light:
image
Content bleed dark:
image
Content bleed paper:
image

Header light:
image
Header dark:
image
Header paper:
image

Highlighted light:
image
Highlighted dark:
image
Highlighted paper:
image

Image light:
image
Image dark:
image
Image paper:
image

Overlay light:
image
Overlay dark:
image
Overlay paper:
image

[WD-11865]: https://warthogs.atlassian.net/browse/WD-11865?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jmuzina** [comment]
FYI, this causes changes to some other examples that used card; the card section of the example now uses the theme of the example, rather than always being light theme.

- [Input on Paper](https://vanilla-framework-5112.demos.haus/docs/examples/base/paper/input-on-paper?theme=paper)
- [Search box on Paper](https://vanilla-framework-5112.demos.haus/docs/examples/base/paper/search-box-on-paper?theme=paper)

**Comment by jmuzina** [comment]
@lyubomir-popov Please give a design review when you have the time!

**Comment by lyubomir-popov** [comment]
Nothing looks broken to me, so +1.  Iwonder whether we need all this though - the closest we come to cards in the rebranded pages are these two examples, and both have no borders or shadows, just the indent. 

image
image
 
Maybe it's time to deprecate all other versions of the card - unless they are used in products:

**Comment by bartaz** [comment]
We "need" this just for the sake of consistency. Current cards would break on dark theme (components used in them would use white text). So we need them to at least be not broken, and use theme colours instead of hardcoded ones.

So it's just a minimal work to achieve that.

But overall at some point we should do a review of Vanilla components and properly identify the ones we want to deprecate.

**Comment by bartaz** [comment]
It seems that border around card is still hardcoded into a single colour, because it's extended out of `%vf-is-bordered` that is using `$border` that is not themed.

**Comment by jmuzina** [comment]
> It seems that border around card is still hardcoded into a single colour, because it's extended out of `%vf-is-bordered` that is using `$border` that is not themed.

@bartaz I've revised this to use `1px solid $colors--theme--border-default`

**Comment by bartaz** [comment]
> > It seems that border around card is still hardcoded into a single colour, because it's extended out of `%vf-is-bordered` that is using `$border` that is not themed.
> 
> @bartaz I've revised this to use `1px solid $colors--theme--border-default`

Thanks, but it would be good to update the `%vf-is-bordered` to be themed as well, as this may fix any other places that use these borders consistently (and that's what we eventually need anyway).

**Comment by bartaz** [comment]
> > update the `%vf-is-bordered` to be themed as well, as this may fix any other places that use these borders consistently (and that's what we eventually need anyway).
> 
> @bartaz Done! Let's see what Percy thinks about this, hopefully many things get fixed

If the colour change is subtle it may not even notice ;)

**Comment by jmuzina** [comment]
> > > update the `%vf-is-bordered` to be themed as well, as this may fix any other places that use these borders consistently (and that's what we eventually need anyway).
> > 
> > 
> > @bartaz Done! Let's see what Percy thinks about this, hopefully many things get fixed
> 
> If the colour change is subtle it may not even notice ;)
@bartaz 
[Percy build is looking good](https://percy.io/bb49709b/vanilla-framework/builds/34703991/changed/1897560182?browser=chrome&browser_ids=56&group_snapshots_by=similar_diff&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280)!

File: templates/docs/examples/patterns/card/header.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 {% block content %}
 <div class="p-card">
-  <img class="p-card__thumbnail" src="https://assets.ubuntu.com/v1/dca2e4c4-raspberry-logo.png" alt="" />
+  <img class="p-card__thumbnail" src="https://assets.ubuntu.com/v1/8fdc8250-Ubuntu+and+Raspberry+Pi.svg" alt="" />
```
Comment (bartaz) [inline templates/docs/examples/patterns/card/header.html:8]: Why the image change?

File: scss/_settings_placeholders.scss [hunk 2]
```diff
@@ -5,7 +5,7 @@
 $input-border-thickness: 1.5px;
 $bar-thickness: 0.1875rem !default; // 3px at 16px fontsize, expressed in rems so it scales with text if the root font-size changes at a breakpoint
 $border-radius: 0; // deprecated, will be removed in future version of Vanilla
-$border: $input-border-thickness solid $color-mid-light !default;
+$border: 1px solid $colors--theme--border-default !default;
```
Comment (bartaz) [inline scss/_settings_placeholders.scss:8]: Keep the border thickness as it was before, just update the colour, please.

File: templates/docs/examples/patterns/card/header.html (hunk 1 above)
Comment (jmuzina) [reply templates/docs/examples/patterns/card/header.html:8]: @bartaz  Updated image looks better on dark theme as it has no dark text

**Review by pastelcyborg — APPROVED** [review]

**Pull request merged**